### PR TITLE
Add Cache Helper

### PR DESCRIPTION
**Pull request by takecare**
We want to be able to cache data that the plugin displays, so we can avoid hammering an API and thus hitting API rate limits.

This PR adds a rather simple cache helper that provides a simple interface, composed of 3 methods:

1. `hasExpired` to check if the stored data is available and out of date;
2. `save` to store data;
3. `read` to read the stored data.

The cache is backed by firebase.

### CODE REVIEW

**Comment by ouchadam** [comment]
Looks good apart from one bit... :palm_tree: The entry point is wrong. 

Having the cache as part of the `plugin-tester` means we won't be able to use the cache in deployed functions since `plugin-tester` is local only.

And we can't pull it out at the moment because the cache has a `require('firebase-admin')` which we can't use from inside a deployed function because we can only call `admin.initialise` once (and injecting dependencies is better).

The entry should be via https://github.com/novoda/dashboards/blob/master/plugin/lib/index.js

We'd end up with something like  

```javascript
module.exports = firebaseDatabase => {
  const generateViewState = plugin.cache(firebaseDatabase, 1000, require('./data-source'))
  // or 
  const cache = plugin.cache(firebaseDatabase)
  const generateViewState = cache.generateViewState(1000, require('./data-source'))

  return plugin.templated(configuration, component, generateViewState)
}
```

File: plugin-tester/src/index.js [hunk 1]
```diff
@@ -1,12 +1,18 @@
 #! /usr/bin/env node
 
+const serviceAccount = require('./firebase-secrets.json');
```
Comment (ouchadam) [inline plugin-tester/src/index.js]: this shouldn't be forced, we should only apply these if a user has specified a `firebase-secrets` via flag

File: plugin-tester/src/runner.js [hunk 2]
```diff
@@ -9,18 +16,19 @@ const ignoredResponse = {
...
                 callbackUrl: `http://localhost:${port}/callback`,
-                configuration: config
+                configuration: config,
+                pluginInstanceId: config.pluginInstanceId
             }
         }
-        decache(path)
-        require(path).plugin()(request, ignoredResponse)
+        decache(pluginPath)
+        require(pluginPath)(request, ignoredResponse)
```
Comment (ouchadam) [inline plugin-tester/src/runner.js]: I may have missed it but this looks like its missing `.plugin(dependencies)`

File: server/src/scheduler/component/plugin-trigger-scheduler-component.js [hunk 3]
```diff
@@ -1,3 +1,43 @@
...
+    )
+}
+
+const tickAllPlugins = pluginInstances => {
+    console.log(pluginInstances)
+    const queryAllPlugins = pluginInstances.map(instance => {
+        const requestPayload = {
+            pluginInstanceId: instance.pluginInstanceId,
+            configuration: instance.configuration,
+            callbackUrl: `https://us-central1-${projectId}.cloudfunctions.net/pluginCallback?pluginInstanceId=${instance.pluginInstanceId}`,
```
Comment (ouchadam) [inline server/src/scheduler/component/plugin-trigger-scheduler-component.js]: this `projectId` needs to come from https://github.com/novoda/dashboards/pull/40/files#diff-da93c66952adaf756f0bc23000673d31R41

File: plugin-tester/src/runner.js (hunk 2 above)
Comment (takecare) [reply plugin-tester/src/runner.js]: it is indeed. i had fixed this but forgot to push.

File: .gitignore [hunk 4]
```diff
@@ -1 +1,2 @@
 **node_modules/
+*secrets.json
```
Comment (ouchadam) [inline .gitignore]: can remove this?

**Comment by ouchadam** [comment]
looking good!  :+1:

**Review by ouchadam — APPROVED** [review]

**Pull request merged**